### PR TITLE
Make the prompts editable now

### PR DESCRIPTION
**Pull request by liwilliam2021**
Adds additional edit functionality that essentially works like deleting and resending a message

### CODE REVIEW

File: src/core/webview/webviewMessageHandler.ts [hunk 1]
```diff
@@ -55,6 +55,140 @@ export const webviewMessageHandler = async (
...
+
+						// Don't initialize with history item for edit operations
+						// The webviewMessageHandler will handle the conversation state
+					}
+				} catch (error) {
+					console.error(`Error in ${operation} message:`, error)
+					vscode.window.showErrorMessage(
+						`Error ${operation === "edit" ? "editing" : "deleting"} message: ${error instanceof Error ? error.message : String(error)}`,
+					)
+				}
```
Comment (daniel-lxs) [inline src/core/webview/webviewMessageHandler.ts:238]: I noticed this function handles both delete and edit operations with quite complex branching logic (lines 65-187). Would it make sense to split this into separate functions like `handleDeleteOperation` and `handleEditOperation` for better maintainability? This could make the code easier to follow and test individually.

File: src/core/webview/__tests__/ClineProvider.spec.ts
Comment (daniel-lxs) [inline src/core/webview/__tests__/ClineProvider.spec.ts:1]: The test coverage looks good for basic edit/delete functionality. Have you considered adding tests for edge cases like:
- Editing messages that contain images or other media
- Handling network failures during edit submission
- Concurrent edit operations

These scenarios could help ensure the feature is robust in real-world usage.

**Review by daniel-lxs — COMMENTED** [review]
This looks good. I left a couple of minor suggestions.

I think Ellipsis is right. Using a separate message for the "Are you sure?" edit dialog makes sense. It might also be a bit confusing to see translations for "delete message" in an edit operation.

**Comment by liwilliam2021** [comment]
> This looks good. I left a couple of minor suggestions.
> 
> I think Ellipsis is right. Using a separate message for the "Are you sure?" edit dialog makes sense. It might also be a bit confusing to see translations for "delete message" in an edit operation.

I think this particular thing should've been resolved in a newer push

**Comment by liwilliam2021** [comment]
@daniel-lxs let me know if the tests are good or getting too long? I tried moving things to a new file but kept running into mock issues

**Comment by daniel-lxs** [comment]
@liwilliam2021 
You can try moving them into a new file if necessary if you are having issues with mocking.

Hit me up if you have any questions!

**Comment by liwilliam2021** [comment]
@daniel-lxs oh sorry! To clarify everything works on my end-- just wasn't sure if src/core/webview/__tests__/ClineProvider.spec.ts was getting too long

**Review by daniel-lxs — APPROVED** [review]
Looks good!
